docroot/modules/custom/sitenow_events/sitenow_events.module

```diff
@@ -648,3 +660,85 @@ function _sitenow_events_build_options_list(array $tree, $r = 0, $p = NULL, arra
 
   return $options;
 }
+
+/**
+ * Implements hook_schemadotorg_jsonld_schema_type_entity_alter().
+ */
+function sitenow_events_schemadotorg_jsonld_schema_type_entity_alter(array &$data, EntityInterface $entity): void {
+  if ($entity instanceof FieldableEntityInterface) {
+    $type = $entity->bundle();
+    if ($type == 'event') {
+
+      // Bring in children events if any.
+      $parent_id = $entity->id();
+      $sub_event_nids = \Drupal::entityQuery('node')
+        ->accessCheck(FALSE)
+        ->condition('type', 'event')
+        ->condition('status', 1)
+        ->condition('field_event_parent', $parent_id, '=')
+        ->execute();
+
+      if ($sub_event_nids) {
+        $sub_events = [];
+        foreach ($sub_event_nids as $nid) {
+          $sub_event = Drupal::entityTypeManager()->getStorage('node')->load($nid);
+          $sub_events[] = [
+            '@type' => 'Event',
+            'name' => $sub_event->label(),
+            '@url' => Url::fromRoute('entity.node.canonical', ['node' => $nid], ['absolute' => TRUE])->toString(),
+          ];
+        }
+        $data['subEvent'] = $sub_events;
+      }
+
+      if ($entity->hasField('field_event_virtual') && !$entity->get('field_event_virtual')->isEmpty()) {
+        $uri = $entity->get('field_event_virtual')->getValue()[0]['uri'];
+        $virtual = [
+          '@type' => 'VirtualLocation',
+          'url' => $uri,
+        ];
+        if (isset($data['location'])) {
+          $data['location'] = [
+            $data['location'],
+            $virtual,
+          ];
+        }
+        else {
+          $data['location'] = $virtual;
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Set dynamic allowed values for the publish_options field.
```

Comment:
This comment seems like its incorrect.

Reply:
This whole function was supposed to be ripped out..